Update decision for the API design (#41)
vshn/application-catalog-docs: docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

* Exposing only needed fields, add more as needed
* Sensible defaults, minimize required parameters.

To minimize confusion when querying the claims.
The kind should also contain the provider as a pre-fix:

.Example Kind
[source, yaml]
----
apiVersion: acmecorp.appcat.vshn.io/v1
kind: AcmecorpPostgreSQL
----

== Rationale

Proposal 2 provides the most flexibility for modelling the various limitations and differences that various providers might have.
All other proposals are hard definitions what the API should look like and are not a convention.
This makes it hard to tailor the various provider limitations and differences to fit the definitions.

To achieve the identity and same look for the various providers and services a flexible convention makes the most sense.

=== Avoiding Confusion with Specific Kinds

> In k8s there's ambiguity if there are multiple CRDs with the same `+kind+` but different `+apiversion+` fields.

Reviewer: ```suggestion
In Kubernetes there's ambiguity if there are multiple CRDs with the same `+kind+` but different `+apiversion+` fields.
```

> Given a K8s cluster with these CRDs:

Reviewer: ```suggestion
Given a Kubernetes cluster with these CRDs:
```

* `+foo.a.io+`
* `+foo.b.io+`

> If a user invokes `+kubectl get foo+` only CRs of type `+foo.a.io+` are returned footnote:ordering[As far as I was able to determine the K8s returns the first API group it finds by alphabetical order.].

Reviewer: ```suggestion
If a user invokes `+kubectl get foo+` only CRs of type `+foo.a.io+` are returned footnote:ordering[As far as we were able to determine, the Kubernetes API server returns the first API group it finds by alphabetical order.].
```

> This could lead to "missing" CRs from a user perspective

Reviewer: ```suggestion
This could lead int to a "missing" CRs situation from a users perspective.
```

Author: > This could lead int to a "missing"

Did you mean "This could lead into a missing"?

* Exposing only needed fields, add more as needed
* Sensible defaults, minimize required parameters.

> To minimize confusion when querying the claims. The kind should also contain the provider as a pre-fix:

Reviewer: IMHO that's much more explanation. When I read that in a few month I don't know anymore what's meant with "To minimize confusion".

Author: I've added more in-depth explanation in the `rationale` section of the decision.

.Example Kind
[source, yaml]
----

> apiVersion: provider.appcat.vshn.io/v1
> kind: ProviderPostgreSQL

Reviewer: ```suggestion
apiVersion: acmecorp.appcat.vshn.io/v1
kind: AcmecorpPostgreSQL
```

I feel "provider" is too generic to understand it
